### PR TITLE
Add createdby_userId during check creation

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
In order to add the new column `createdby_userId` and populate it as part of the check creation process the type needs to be dropped and the `spCreateChecks` procedure recreated.

### CODE REVIEW

File: admin/services/check-start.service/check-start.service.js [hunk 1]
```diff
@@ -194,6 +201,7 @@ checkStartService.initialisePupilCheck = async function (
   availableForms,
   usedFormIds,
   isLiveCheck,
+  userId,
   schoolId = null,
```
Comment (GeorgeChatzigiannis) [inline admin/services/check-start.service/check-start.service.js]: Not relevant to my PR but this seems to be wrong. `schoolId` will definitely not be null at that stage (the caller method already evaluates that) so providing a default to null seems to be totally wrong here.

Comment (GuyHarwood) [reply admin/services/check-start.service/check-start.service.js]: please check existing call sites and remove default if not required

File: admin/data/sql/migrations/20200318120158.do.update-check-add-created-by-user-id.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE [mtc_admin].[check]
+ADD createdBy_userId INT NULL
+CONSTRAINT [FK_check_createdBy_userId] FOREIGN KEY([createdBy_userId])
+REFERENCES [mtc_admin].[user] ([id])
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20200318120158.do.update-check-add-created-by-user-id.sql]: Missing semi-colon

File: admin/data/sql/migrations/20200318120158.undo.update-check-add-created-by-user-id.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE [mtc_admin].[check]
+DROP CONSTRAINT [FK_check_createdBy_userId];
+ALTER TABLE [mtc_admin].[check]
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20200318120158.undo.update-check-add-created-by-user-id.sql]: Is this okay like this?  Or should the two ALTERs go in separate files?

File: admin/data/sql/migrations/20200318133108.do.drop-spcreatechecks.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+DROP PROCEDURE IF EXISTS [mtc_admin].[spCreateChecks];
```
Comment (jon-shipley) [inline admin/data/sql/migrations/20200318133108.do.drop-spcreatechecks.sql]: I'm surprised this is needed - you are using the CREATE OR ALTER statement for the new definition.

Comment (GuyHarwood) [reply admin/data/sql/migrations/20200318133108.do.drop-spcreatechecks.sql]: i would leave it in, as it makes the operation fail safe.  although there is the argument 'surely it should fail if it's not there as there would be a wider issue?'

File: admin/data/sql/migrations/20200318120158.undo.update-check-add-created-by-user-id.sql (hunk 3 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20200318120158.undo.update-check-add-created-by-user-id.sql]: Looks ok to me. they can happen in a single transaction.

File: admin/data/sql/migrations/20200318133108.do.drop-spcreatechecks.sql (hunk 4 above)
Comment (jon-shipley) [reply admin/data/sql/migrations/20200318133108.do.drop-spcreatechecks.sql]: It won't fail if it already exists or if it doesn't exist - it's using CREATE OR ALTER PROCEDURE

**Pull request merged**